### ISSUE TITLE
Capture not working correctly

### ISSUE THREAD

**Issue by fzakaria** [jonatas/fast#18]
Hi !

I'm trying to match the following AST examples:

```
 a.must_equal true, "error"

(send
  (lvar :a) :must_equal
  (true)
  (str "error))
```

I can capture the AST correctly with `fast '(send _ :must_equal _)'` however I'd like to capture the `a` variable.

I've tried adding **$** according to the documentation such as:
`'(send $_ :must_equal _)'` however that returns a lot of additional AST nodes.

The ultimate goal I'd like to accomplish is to turn the ruby snippet to:

```
expect(a).must_equal true, "error"
```

**Bonus points:** `a` may be a object property like `a.b.c` or `Module.a` & I would like to capture the whole chain.

**Comment by fzakaria** [comment]
The library is pretty cool though.

**Comment by fzakaria** [comment]
I opened an issue to Minitest library https://github.com/seattlerb/minitest/pull/828/files demonstrating how it can be used to help migrate others to Minitest 6!

I couldn't rely on _search_all_ because our codebase is JRuby and I was seeing a `Unhandled Java exception: org.jruby.RubyContinuation$Continuation` error

**Comment by jonatas** [comment]
Thanks for reporting @fzakaria!

Here is a simple example showing how to use the properly:

```ruby
⋊> ~/c/fast on master ⨯ bin/console                                                                            12:11:05
Resolving dependencies...
[1] pry(main)> ast = Fast.ast('a.must_equal true, "error"')
=> s(:send,
  s(:send, nil, :a), :must_equal,
  s(:true),
  s(:str, "error"))
[2] pry(main)> Fast.capture('(send (send nil $_) _)', ast)
=> [:a]
```

Keep in mind that `_` means anything but it does not mean digging into the node.

Looking for your example of replacing using the capture in the block you can do:

```ruby
Fast.replace('(send (send nil $_) _)', ast) { |node, capture|  replace(node.children[0].loc.expression, "expect(#{capture})")}
=> "expect([:a]).must_equal true, \"error\""
```

**Comment by jonatas** [comment]
> I couldn't rely on search_all because our codebase is JRuby and I was seeing a Unhandled Java exception: org.jruby.RubyContinuation$Continuation error

About it, I'm not sure because probably we need to make parser work with JRuby to make it easy.

**Comment by fzakaria** [comment]
@jonatas thanks for looking into it.
I had one more question:

I noticed I put multiple `Fast.rewrite_file` in a single   Fastfile shortcut.
Such as:
```
test_files = Fast.ruby_files_from('test')

test_files.each do |file|

Fast.rewrite_file('...', file) do |node|
end

Fast.rewrite_file('...', file) do |node|
end
end
```

The files would end up corrupted or it would make very weird rewrites that were not clear.
I solved it by breaking up the rewrites into their own shortcut but it looks like maybe the files are not flushing in between.
